File: src/main/java/org/cimsbioko/syncadpt/AuthenticatorService.java
package org.cimsbioko.syncadpt;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;

public class AuthenticatorService extends Service {

    private AccountAuthenticator auth;

    @Override
    public void onCreate() {
        auth = new AccountAuthenticator(this);
    }

    @Override
    public IBinder onBind(Intent intent) {
        return auth.getIBinder();
    }
}

File: src/main/java/org/cimsbioko/syncadpt/AuthenticatorService.kt
package org.cimsbioko.syncadpt

import android.app.Service
import android.content.Intent
import android.os.IBinder

class AuthenticatorService : Service() {

    private lateinit var auth: AccountAuthenticator

    override fun onCreate() {
        auth = AccountAuthenticator(this)
    }

    override fun onBind(intent: Intent): IBinder {
        return auth.iBinder
    }
}